Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -276,24 +303,47 @@ private boolean deletePoint(NodeReference nodeReference, Point point, long seque
                 setParent(rootIndex, null);
             } else {
                 replaceNodeBySibling(grandParent, parent, nodeReference);
+                updateAncestorNodesAfterDelete(grandParent, point);
             }
             delete(nodeReference);
             delete(parent);
-            return false;
+            return;
```

Comment:
minor: unnecessary return statement

Reply:
The else part was/is not present. We do not want the subsequent block evaluated.

Reply:
Oops, I was thinking that this was the last statement in a function call (and not the last statement in an interior code block).